server/webapp/WEB-INF/rails.new/app/controllers/api_v2/agents_controller.rb

```diff
@@ -33,13 +33,16 @@ def show
 
     def update
       result    = HttpOperationResult.new
-      agent_service.updateAgentAttributes(current_user, result, params[:uuid], params[:hostname], maybe_join(params[:resources]), maybe_join(params[:environments]), to_enabled_tristate)
+
+      agent = agent_service.updateAgentAttributes(current_user, result, params[:uuid], params[:hostname], maybe_join(params[:resources]), maybe_join(params[:environments]), to_enabled_tristate)
 
       if result.isSuccess
         load_agent
         render json_hal_v2: agent_presenter.to_hash(url_builder: self)
       else
-        render_http_operation_result(result)
+        @agent_instance = agent
```

Comment:
This feels a bit redundant, if the `@agent_instance` instance variable were set on line 37 above, at worst it'll be nil.